### PR TITLE
Feat/125 extend assets target path

### LINKED ISSUE #125: [Feature Request]: allow single files in build and push workflow as a target path
### Is your feature request related to a problem?

I would like to be able to do:
```
ASSETS_TARGET_PATHS: './assets ./my-file.txt'
```

Currently the workflow is creating directories prior to the build see [here](https://github.com/inpsyde/reusable-workflows/blob/main/.github/workflows/build-and-push-assets.yml#L219).

And then instead of doing `git add ${path}` is doing `git add -f "${path}/*`

### Describe the desired solution

We could use something like:
```
ASSETS_TARGET_PATHS: './assets/ ./my-file.json'
```
If the path ends in `/` then it is a directory else a file. 

This is a breaking change but it is more flexible.

### Describe the alternatives that you have considered

I considered the following:

- Creating ASSETS_TARGET_PATHS_FILES: But I think it is not better than one single place to declare target paths.

### Additional context

We could change the default 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by luislard**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Fixes #125  

**What is the current behavior?** (You can also link to an open issue here)
`ASSETS_TARGET_PATHS` do not allow file paths, only directory paths.

**What is the new behavior (if this is a feature change)?**
We can refer to files using a new input.

Example:
```
ASSETS_TARGET_FILES: './a-file.txt ./another-file.txt'
```

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

**Other information**:

### CODE REVIEW

**Comment by luislard** [comment]
We delivered an approach of supporting folders and files:

**How was it working**
We can refer to directories or files.

- If the path ends with a slash is treated as a folder. We will create it if it doesn't exist.
- If the path do not end with a slash is treated as a file. We will create it if it doesn't exist.
Example:
```
ASSETS_TARGET_PATHS: './a-folder/ ./another-folder/ ./a-file.txt'
```
**The Problem**
Was introducing a BC. Current consumers need to update to add the forward slash to each path. It is a quick change, though.

We agreed in ternally to use another VAR like explained in the options considered.
[Commit for the first iteration](https://github.com/inpsyde/reusable-workflows/blob/6596483482e46c135efeae03e6c1238796b3ffdb/.github/workflows/build-and-push-assets.yml)

**Review by shvlv — APPROVED** [review]
Looks good to me!

File: .github/workflows/build-and-push-assets.yml [hunk 1]
```diff
@@ -44,10 +44,15 @@ on:
         required: false
         type: string
       ASSETS_TARGET_PATHS:
-        description: Target path(s) for compiled assets.
+        description: Target directory path(s) for compiled assets.
```
Comment (tyrann0us) [inline .github/workflows/build-and-push-assets.yml:47]:
```suggestion
        description: Space-separated list of target directory paths for compiled assets.
```

File: .github/workflows/build-and-push-assets.yml [hunk 2]
```diff
@@ -44,10 +44,15 @@ on:
...
         default: './assets'
         required: false
         type: string
+      ASSETS_TARGET_FILES:
+        description: Target file path(s) for compiled assets.
```
Comment (tyrann0us) [inline .github/workflows/build-and-push-assets.yml:52]:
```suggestion
        description: Space-separated list of target file paths for compiled assets.
```

File: docs/build-and-push-assets.md [hunk 3]
```diff
@@ -107,7 +107,8 @@ This is not the simplest possible example, but it showcases all the recommendati
 | `COMPILE_SCRIPT_PROD` | `'build'`                     | Script added to `npm run` or `yarn` to build production assets                                                                  |
 | `COMPILE_SCRIPT_DEV`  | `'build:dev'`                 | Script added to `npm run` or `yarn` to build development assets                                                                 |
 | `MODE`                | `''`                          | Mode for compiling assets (`prod` or `dev`)                                                                                     |
-| `ASSETS_TARGET_PATHS` | `'./assets'`                  | Target path(s) for compiled assets                                                                                              |
+| `ASSETS_TARGET_PATHS` | `'./assets'`                  | Target directory path(s) for compiled assets                                                                                    |
+| `ASSETS_TARGET_FILES` | `''`                          | Target file path(s) for compiled assets                                                                                         |
```
Comment (tyrann0us) [inline docs/build-and-push-assets.md:111]:
```suggestion
| `ASSETS_TARGET_PATHS` | `'./assets'`                  | Space-separated list of target directory paths for compiled assets                                                              |
| `ASSETS_TARGET_FILES` | `''`                          | Space-separated list of target file paths for compiled assets                                                                   |
```

File: docs/build-and-push-assets.md [hunk 4]
```diff
@@ -173,9 +174,11 @@ jobs:
       MODE: ${{ github.ref_type == 'branch' && github.ref_name == 'production' && 'prod' || '' }}
 ```
 
-> Can I have multiple output folders for my package?
+> Can I have multiple output folders for my package?, and what about files?
```
Comment (tyrann0us) [inline docs/build-and-push-assets.md:177]: `?,` seriously? 🤪
```suggestion
> Can I have multiple output folders for my package? What about files?
```

File: docs/build-and-push-assets.md [hunk 5]
```diff
@@ -173,9 +174,11 @@ jobs:
...
 
-Yes, the `inputs.ASSETS_TARGET_PATHS` accepts multiple space-separated paths:
+Yes, 
+- The `inputs.ASSETS_TARGET_PATHS` accepts multiple space-separated paths for **directories**
+- The `inputs.ASSETS_TARGET_FILES` accepts multiple space-separated paths for **files**
```
Comment (tyrann0us) [inline docs/build-and-push-assets.md:181]:
```suggestion
Yes, `inputs.ASSETS_TARGET_PATHS` and `inputs.ASSETS_TARGET_FILES` accept multiple space-separated paths for directories and files, respectively.
```

**Review by tyrann0us — APPROVED** [review]
Thank you for incorporating the changes. Haven't tested this but code-wise LGTM.

**Pull request merged**